app/controllers/podcasts_controller.rb

```diff
@@ -1,7 +1,15 @@
 class PodcastsController < ApplicationController
 
   def index
-    @podcasts = Podcast.all 
+    if params[:tags].present?
+      @podcasts = Podcast.joins(:tags).where(tags: {id: params[:tags][:ids]})
+    else
+      @podcasts = Podcast.where(nil)
+    end
+    @podcasts = @podcasts.filter_by_active
+    @podcasts = @podcasts.filter_by_location(params[:location])if params[:location].present?
+    @podcasts = @podcasts.filter_by_adult_content(params[:adult_content])if params[:adult_content].present?
+    @podcasts = @podcasts.filter_by_name(params[:name])if params[:name].present?
```

Comment:
links together the different filters that can be applied and narrows down results with each filter. This could definitely be a candidate for refactor.

Reply:
This all looks great! Is this something you think we could move into a facade to save some space in the controller?